### DISCUSSION TITLE
Correct way to redirect.

### DISCUSSION THREAD

**Discussion by mahmutbayri** [clue/framework-x#106] [Q&A] [answered]
Hi,
I've created a simple crud app. After handling the request (delete, update, store data) i want to redirect user to a specific page.

In order to do that i used this way, because RedirectHandler doesn't work what i expected.

This is a part of my `DestroyTaskController`

```php
//// ....
public function __invoke(ServerRequestInterface $request)
{
    $id = $request->getAttribute('id');
    return sqliteConnection()
        ->query(
            'DELETE FROM tasks WHERE id = ?',
            [$id]
        )
        ->then(function (Result $result) {
            return new Response(
                301,
                [
                    'Location' => '/tasks'
                ],
                $result->changed ? 'deleted' : 'not found'
            );
        });
}
//// ....
```

The repository is here -> https://github.com/mahmutbayri/framework-x-database-crud

My question is whether we have a cleaner method.

Thanks

**Comment by SimonFrings** [comment, marked as answer]
@mahmutbayri Your code looks fine, nothing wrong with that :+1: 

@clue and I have a similar approach as #76 in mind. We think that this should be part of [reactphp/http](https://github.com/reactphp/http) and look something like this: `React\Http\Message\Response::redirect($url)`.

Something is coming up soon 🔥
